Controllers/frontend/NostoTagging.php

```diff
@@ -330,4 +331,28 @@ private function fireRequest(NostoOAuthToken $token)
         }
         return $result;
     }
+
+    /**
+     * Restores the cart based on the given hash
+     *
+     * @throws Enlight_Event_Exception
+     * @throws Enlight_Exception
+     * @throws Zend_Db_Adapter_Exception
+     * @throws Exception
+     */
+    public function cartAction()
+    {
+        $hash = $this->Request()->getParam(CartRestore::CART_RESTORE_URL_PARAMETER);
+        $sessionId = Shopware()->Session()->get('sessionId');
+        if ($hash === null || $sessionId === null) {
+            return;
```

Comment:
We should always redirect user to somewhere. Perhaps front page in this case

Reply:
Fixed, thanks